plugins/survey/src/types.ts

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright © 2023 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { AttachedDoc, Class, CollectionSize, Doc, Markup, Ref, Space } from '@hcengineering/core'
+import { Attachment } from '@hcengineering/attachment'
+import { Person } from '@hcengineering/contact'
+import { ComponentType, SvelteComponent } from 'svelte'
+import { Resource } from '@hcengineering/platform'
+
+/**
+ * @public
+ *
+ * @link https://www.npmjs.com/package/lexorank
+ */
+export type Rank = string
+
+/**
+ * ∈ [0, 1]
+ */
+export type Fraction = number
+
+/** @public */
+export interface Survey extends Space {
+  name: string
+  questions: CollectionSize<Question>
+  requests: CollectionSize<Request>
+  // TODO: isAssessment: boolean
+}
+
+/** @public */
+export interface Question extends AttachedDoc<Survey, 'questions', Survey> {
+  title: Markup
+  rank: Rank
+  attachments: CollectionSize<Attachment>
+  assessment?: AssessmentData<this>
+}
+
+/** @public */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export interface AssessmentData<Q extends Question> {
+  weight: number
+}
+
+/** @public */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unused-vars
+export interface AnswerData<Q extends Question> {}
+
+/** @public */
+export interface QuestionOption {
+  label: string
+}
+
+/** @public */
+export interface SingleChoiceQuestion extends Question {
+  options: QuestionOption[]
+  shuffle: boolean
+}
+
+/** @public */
+export interface SingleChoiceAnswerData extends AnswerData<SingleChoiceQuestion> {
+  selection?: number
```

Comment:
1. Is it the selected option index? Can we make it clear by making the name more specific or adding a comment?
2. It might become complicated to work with indices when `shuffle` option is on. If I'm getting it right it will shuffle the options for the trainee so the indices in his training UI will be different from the indices in the original question. We may also need to store these indices so once they are shuffled for one trainee they are stable every time he opens his training. So it might be confusing which index this answer is referring to.